TradingBot/Services/Strategy.cs

```diff
@@ -1,33 +1,188 @@
     {
         var predictions = await _predictor.GetPredictionsAsync();
         var assets = await _assetsDataSource.GetAssetsAsync();
-        return new[]
+        var tradingActions = new List<TradingAction>();
+
+        var sellActions = new List<TradingAction>();
+        var growthRates = new List<averageGrowthRate>();
+
+        var cashAvailable = assets.Cash.BuyingPower;
+
+        foreach (var prediction in predictions)
         {
-            TradingAction.MarketBuy(new TradingSymbol("TSLA"), 0.5m, _clock.UtcNow)
-        };
+            var symbol = prediction.Key;
+            var closingPrices = new List<decimal>();
+            var currentPrice = await GetCurrentPrice(symbol, token);
+            if (currentPrice != null)
+                closingPrices.Add(currentPrice.Close);
+
+            closingPrices.AddRange(prediction.Value.Prices.Select(dailyPrice => dailyPrice.ClosingPrice).ToList());    
+
+            if (IsPriceDecreasing(closingPrices, minDaysDecreasing) && assets.Positions.TryGetValue(symbol, out var position))
+            {
+                if (position.AvailableQuantity > 0)
+                    sellActions.Add(TradingAction.MarketSell(symbol, position.AvailableQuantity, _clock.UtcNow));
+            }
+            else if (IsPriceIncreasing(closingPrices, minDaysIncreasing))
+            {
+                decimal growthRate = CalculateAverageGrowthRate(closingPrices);
+                var lowPrice0 = prediction.Value.Prices.Select(dailyPrice => dailyPrice.LowPrice).ToList()[0];
+                var lowPrice1 = prediction.Value.Prices.Select(dailyPrice => dailyPrice.LowPrice).ToList()[1];
+                var buyLimitPrice = lowPrice0 + (lowPrice1 - lowPrice0) / 2;
+                if(currentPrice != null)
+                {
+                    var tomorrowLowPrice = lowPrice0;
+                    if (tomorrowLowPrice > currentPrice.Low)
+                        buyLimitPrice = currentPrice.Low + (tomorrowLowPrice - currentPrice.Low) / 2;
+                    else
+                        buyLimitPrice = tomorrowLowPrice + (currentPrice.Low - tomorrowLowPrice) / 2;
+                }
+
+                growthRates.Add(new averageGrowthRate(symbol, growthRate, buyLimitPrice));
+            }
+        }
+
+        tradingActions.AddRange(sellActions);
+        tradingActions.AddRange(GetBuyActions(growthRates, cashAvailable, maxBuyCount, topGrowingSymbolsBuyRatio));
+
+        return tradingActions;
+    }
+
+    private async Task<DailyTradingData?> GetCurrentPrice(TradingSymbol symbol, CancellationToken token = default)
+    {
+        var today = DateOnly.FromDateTime(_clock.UtcNow.UtcDateTime);
+        var result = await _marketDataSource.GetDataForSingleSymbolAsync(symbol, today, today, token);
+
+        return (result == null) ? null : result.First();
+    }
```

Comment:
If we got prediction about some symbol, we should be able to retrieve it's last price. So if we get `null`, we should throw some exception (maybe `UnreachableException`, since it should never logically happen?).

Reply:
Also, this assumes that we _can_ get today's data. If we do prediction before market opens, I _think_ we would just get empty list here? So IMO we should rename this to `GetLastAvailablePriceAsync()`, and change it a little bit to do just that.